src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_client.py

```diff
@@ -57,7 +57,15 @@ def _send(self, method, url, param=None, body=None, api_version=V1_0):
                 r = send_raw_request(self._cli_ctx, method, url, resource=self._resource, uri_parameters=param,
                                      body=body)
             except HTTPError as ex:
-                raise GraphError(ex.response.json()['error']['message'], ex.response) from ex
+                err = ex.response.json()['error']
+                code = err['code']
+                message = err['message']
+                graph_error_message = (
+                    f"({code}) {message}\n"
+                    f"Code: {code}\n"
+                    f"Message: {message}"
```

Comment:
I am hesitating on these 2 lines as they are simply repetitions of the first line and gives no more information, such as

```
(CannotUpdateLockedServicePrincipalProperty) Property passwordCredentials is invalid.
Code: CannotUpdateLockedServicePrincipalProperty
Message: Property passwordCredentials is invalid.
```

ARM clients' implementation is from `azure.core.exceptions.ODataV4Format`:

https://github.com/Azure/azure-sdk-for-python/blob/9da614d54c97660b7f14854a4bce3704242507eb/sdk/core/azure-core/azure/core/exceptions.py#L248-L270

```py
    def __str__(self) -> str:
        return "({}) {}\n{}".format(self.code, self.message, self.message_details())


    def message_details(self) -> str:
        """Return a detailed string of the error.

        :return: A string with the details of the error.
        :rtype: str
        """
        error_str = "Code: {}".format(self.code)
        error_str += "\nMessage: {}".format(self.message)
        if self.target:
            error_str += "\nTarget: {}".format(self.target)


        if self.details:
            error_str += "\nException Details:"
            for error_obj in self.details:
                # Indent for visibility
                error_str += "\n".join("\t" + s for s in str(error_obj).splitlines())


        if self.innererror:
            error_str += "\nInner error: {}".format(json.dumps(self.innererror, indent=4))
        return error_str
```